### PR TITLE
documentation improvements, slight code changes to accommodate hooks module

### PR DESCRIPTION
**Pull request by jhallard**
The documentation has been updated slightly to be more clean and explicit. I also added the `--hook_data_json` argv item to allow a user to pass in arbitrary data through to the hook module. Simply submit a valid json dictionary to this argument, we will read it into a python dictionary and forward that to the `set_hook_data()` function inside the hook module (should it exist) before we do anything else with the module. The logger is also passed in so that the hook module can use the same logger as the importer script.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,28 +23,41 @@ which tuna or marlin was brought on board as in [this demo](https://www.youtube.
...
+                        regex to find camera name
   -c, --csv             dump csv log file
   -s STORAGE, --storage STORAGE
                         location of the local storage db
   -f FOLDER, --folder FOLDER
-                        folder (directory) to process
-  -i HOST, --host HOST  the segmenter's IP address
-  -p PORT, --port PORT  the segmenter's port number
+                        folder to process
+  -i HOST, --host HOST  the segmenter ip
```
Comment (jcmaslan) [inline README.md]: @jhallard is this limited to IP address or can it also be a host name on the local network network?

File: README.md [hunk 2]
```diff
@@ -23,28 +23,41 @@ which tuna or marlin was brought on board as in [this demo](https://www.youtube.
...
+  -p PORT, --port PORT  the segmenter port number
   -m HOOK_MODULE, --hook_module HOOK_MODULE
-                        hook module for custom functions
-
+                        full path to hook module for custom functions (a
+                        python file)
+  -d HOOK_DATA_JSON, --hook_data_json HOOK_DATA_JSON
+                        a json-dictionary containing extra information to be
```
Comment (jcmaslan) [inline README.md]: Is it necessarily a dictionary? Isn't it just an JSON Object?

File: README.md [hunk 3]
```diff
@@ -135,11 +147,36 @@ def post_video_content(host, port, camera_name, camera_id, filepath, timestamp,
         camera_id   - the ID of the camera as returned from the register_camera function
         filepath    - full path to the video file that needs segmentation
         timestamp   - the earliest timestamp contained in the video file
-        latlng (opt) - the lat/long of the camera (as parsed from the filename)
+        location (opt) - a string describing the location of the camera (example lat-long)
```
Comment (jcmaslan) [inline README.md]: please make example concrete.

File: README.md [hunk 4]
```diff
@@ -135,11 +147,36 @@ def post_video_content(host, port, camera_name, camera_id, filepath, timestamp,
...
     
     description: this function is called each time the importer finds a video for a specific camera. 
                  Put the code for posting the video and meatadata to the segmenter inside this function.
     """
     pass
 ```
+
+#### Set Hook Data Function
+
+Sometimes there is extra data that the hook-module needs from the user but is not explicitely defined in the importer arguments,
```
Comment (jcmaslan) [inline README.md]: explicitly

File: README.md [hunk 5]
```diff
@@ -135,11 +147,36 @@ def post_video_content(host, port, camera_name, camera_id, filepath, timestamp,
...
+you can pass this data in using the `--hook_data_json` argument. Simple give something like 
+
+```
+python importer "... some arguments here..." --hook_data_json '{"camera_plan": "pro", "user_id": "AABBCCDD"}'
+```
+
+And the json-dictionary that you define will be de-serialized into a python dictionary and passed to the `set_hook_data` function defined
```
Comment (jcmaslan) [inline README.md]: Same question about dictionary vs. any object at all?

File: importer.py [hunk 6]
```diff
@@ -40,11 +41,11 @@ def get_params(self, path):
                     lng = float(match.group('lng'))
                 except: pass
         if not camera:
-            print 'did not detect camera name, assuming "default"'
+            logging.warn('did not detect camera name, assuming "default"')
```
Comment (jcmaslan) [inline importer.py]: If this is a name placeholder, how about "Unnamed" instead so convey that it's not related to a system default.

**Review by jcmaslan — APPROVED** [review]
LGTM with changes requested

File: README.md (hunk 1 above)
Comment (jhallard) [reply README.md]: It does not have to be an IP address, good point. Will update to reflect such.

File: README.md (hunk 2 above)
Comment (jhallard) [reply README.md]: you're right, my terminology is mixed up. It is just an object.

File: README.md (hunk 3 above)
Comment (jhallard) [reply README.md]: where did my example go? It was here... will fix with concrete example

Comment (jhallard) [reply README.md]: got it, the example was: 

```json
{"location": {"lat": 7.367598, "lng":134.706975}, "accuracy":5.0}
```

Comment (mdipierro) [reply README.md]: oops. sorry. Perhaps I made a merge error.

**Comment by mdipierro** [comment]
LGTM

**Pull request merged**